cmd/update_selector.go

```diff
@@ -0,0 +1,64 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/syntasso/kratix/api/v1alpha1"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/yaml"
+	"strings"
+)
+
+var updateSelectorCmd = &cobra.Command{
+	Use:   "selector KEY:VALUE",
+	Short: "Command to update destination selectors",
+	Long:  "Command to update destination selectors",
+	Example: `  # add a new destination selector
+  kratix update selector env:dev
+  # removes an existing destination selector
+  kratix update selector zone-
+`,
+	RunE: UpdateSelector,
+	Args: cobra.ExactArgs(1),
+}
+
+func init() {
+	updateCmd.AddCommand(updateSelectorCmd)
+	updateSelectorCmd.Flags().StringVarP(&dir, "dir", "d", ".", "Directory to read Promise from")
+}
+
+func UpdateSelector(cmd *cobra.Command, args []string) error {
+	var promise v1alpha1.Promise
+	if promise, err = getPromise(filepath.Join(dir, "promise.yaml")); err != nil {
+		return fmt.Errorf("failed to find promise.yaml in directory: %v", err)
+	}
+
+	parsed := strings.Split(args[0], ":")
```

Comment:
a similar logic exists in the update_api.go... there may be a refactoring opportunity here 🤔

Reply:
I'm reluctant to do that refactoring. Even though the syntax is the same, they are completely different object: one represents property name and type for CRD and one is key and value for a label. I chose to follow the same syntax for simplicity but it would seem odd to couple them together.

Reply:
> I'm reluctant to do that refactoring. Even though the syntax is the same, they are completely different object: one represents property name and type for CRD and one is key and value for a label. I chose to follow the same syntax for simplicity but it would seem odd to couple them together.

👍🏽 

> Perhaps I could go with kratix update selector key=value? Do you think that syntax will fit labels better?
its a good point; this is actually assigning a value... so yeah, `key=value` may be better.